lib/jsdom/living/fetch/Headers-impl.js

```diff
@@ -0,0 +1,231 @@
+    }
+  }
+
+  _fill(init) {
+    if (Array.isArray(init)) {
+      for (const header of init) {
+        if (header.length !== 2) {
+          throw new TypeError("init is invalid");
+        }
+        this.append(header[0], header[1]);
+      }
+    } else {
+      for (const key of Object.keys(init)) {
+        this.append(key, init[key]);
+      }
+    }
+  }
+
+  has(name) {
+    assertName(name);
+    return this.headersList.has(name.toLowerCase());
+  }
+
+  get(name) {
+    assertName(name);
+    const values = this.headersList.get(name.toLowerCase());
+    if (!values) {
+      return null;
+    }
+    return values.map(h => h.value).join(", ");
+  }
+
+  _removePrivilegedNoCORSHeaders() {
+    this.headersList.delete("range");
+  }
+
+  _append(name, value) {
+    const existing = this.headersList.get(name.toLowerCase());
+    if (existing) {
+      name = existing[0].name;
+      existing.push({ name, value });
+    } else {
+      this.headersList.set(name.toLowerCase(), [{ name, value }]);
+    }
+  }
+
+  append(name, value) {
+    value = value.trim();
+    assertName(name);
+    assertValue(value);
+
+    switch (this.guard) {
+      case "immutable":
+        throw new TypeError("Headers is immutable");
+      case "request":
+        if (isForbiddenHeader(name)) {
+          return;
+        }
+        break;
+      case "request-no-cors": {
+        let temporaryValue = this.get(name);
+        if (temporaryValue === null) {
+          temporaryValue = value;
+        } else {
+          temporaryValue += `, ${value}`;
+        }
+        if (!isCORSWhitelisted(name, value)) {
+          return;
+        }
+        break;
+      }
+      case "response":
+        if (FORBIDDEN_RESPONSE.has(name.toLowerCase())) {
+          return;
+        }
+        break;
+    }
+
+    this._append(name, value);
+    this._removePrivilegedNoCORSHeaders();
+  }
+
+  set(name, value) {
+    value = value.trim();
+    assertName(name);
+    assertValue(value);
+
+    switch (this.guard) {
+      case "immutable":
+        throw new TypeError("Headers is immutable");
+      case "request":
+        if (isForbiddenHeader(name)) {
+          return;
+        }
+        break;
+      case "request-no-cors": {
+        if (!isCORSWhitelisted(name, value)) {
+          return;
+        }
+        break;
+      }
+      case "response":
+        if (FORBIDDEN_RESPONSE.has(name.toLowerCase())) {
+          return;
+        }
+        break;
+    }
+    this.headersList.set(name.toLowerCase(), [{ name, value }]);
+    this._removePrivilegedNoCORSHeaders();
+  }
+
+  delete(name) {
+    assertName(name);
+
+    switch (this.guard) {
+      case "immutable":
+        throw new TypeError("Headers is immutable");
+      case "request":
+        if (isForbiddenHeader(name)) {
+          return;
+        }
+        break;
+      case "request-no-cors": {
+        if (
+          !NO_CORS_SAFELISTED_REQUEST.has(name.toLowerCase()) &&
```

Comment:
Yeah, making these functions instead of sets would be much nicer, so you didn't have to remember to lowercase every time.